### ISSUE TITLE
Is there a way to show only social authentication using the Authenticator widgets?

### ISSUE THREAD

**Issue by sowens-csd** [aws-amplify/amplify-flutter#3456]
I'm using Cognito for social authentication but don't want to allow the user to create their own account. They should instead use one of the configured social authentication services. I haven't been able to find a way to configure it to disable the Cognito authentication and display just the social login buttons. I know I could recreate the buttons but before I did that I wanted to check if I was missing something obvious.

**Comment by Equartey** [comment]
Hey @sowens-csd, 

Can you provide more context of your environment? 

Are you using our [Authenticator project](https://ui.docs.amplify.aws/flutter/connected-components/authenticator)?

**Comment by sowens-csd** [comment]
Yes. I have a working Flutter app using the Authenticator widget from:
```dart
import 'package:amplify_authenticator/amplify_authenticator.dart';
```
 
That successfully displays the login form and if I select one of the social buttons the login is successful. I'd like to be able to customize the displayed form to only display the social login buttons, or have access to those buttons so I could use them on my own form.

**Comment by Equartey** [comment]
Hey @sowens-csd,

While we do offer methods to override the login form, [reference to example](https://github.com/aws-amplify/amplify-flutter/blob/34f0f42c2883d761cb69026b57a23a7a765053cf/packages/authenticator/amplify_authenticator/example/lib/customization/authenticator_with_custom_layout.dart). I'm not seeing an easy way to implement a view using only social sign in buttons. 

This seems to be a shortcoming with the types we are exposing under the hood. I'll continue to investigate and for now will mark this as a feature request.

**Comment by sowens-csd** [comment]
Sounds good, thanks for following up. I'll implement that myself then, just wanted to make sure I wasn't missing the easy way. Would you like to leave this open to track the feature request?

**Comment by minerasoft** [comment]
Hello, any updates on this? It would be very convenient to have the option to enable only social auth in the SignInForm. Or an alternate way to support the same requirement. Thanks

**Comment by Equartey** [comment]
Hi @minerasoft,

No updates at the moment, but it's good to see more interest on this issue. We will continue to monitor and update this issue as we have updates. Thanks

**Comment by dropik** [comment]
+1 for this feature request. The thing is that it's not just about buttons. But it's much more about keeping access/refresh tokens, and delegating all the auth flow logic to Amplify without reimplementing it by yourself. So that would be really great to see an easy option to disable standard login mechanism. Thank you.

**Comment by DarrylSSY** [comment]
Would love this feature too

**Comment by mtliendo** [comment]
If it helps anyone, I use the `Authenticator` component and throw in the following css:

```css
/* Hide everything after the Google sign-in container */
form[data-amplify-form] > .amplify-flex:not(.federated-sign-in-container) {
	display: none !important;
}

/* Optionally hide the divider that says "or" */
form[data-amplify-form] .amplify-divider {
	display: none !important;
}
```

**Comment by markvp** [comment]
it seems like this would be relatively simple...

```
class SignInForm extends AuthenticatorForm {
  /// {@macro amplify_authenticator.sign_in_form}
  SignInForm({super.key, super.includeDefaultSocialProviders})
    : super._(
        fields: [SignInFormField.username(), SignInFormField.password()],
        actions: const [SignInButton(), ForgotPasswordButton()],
      );

  /// A custom Sign In form.
  const SignInForm.custom({
    super.key,
    required super.fields,
    super.includeDefaultSocialProviders,
  }) : super._(actions: const [SignInButton(), ForgotPasswordButton()]);

  @override
  AuthenticatorFormState<SignInForm> createState() => _SignInFormState();
}
```

would it not be possible to expose something like a `SignInForm.socialOnly` which passes an empty array of fields and actions?

**Comment by markvp** [comment]
further to this, if you wanted to avoid the tab for Sign In | Sign Up - there would need to be some way of passing a variable to the authenticator screen to either not set these:

```
const signInUpTabs = [AuthenticatorStep.signIn, AuthenticatorStep.signUp];
```

or adjust this...

```
          child: AuthenticatorTabView(
            tabs: signInUpTabs,
            initialIndex: step == AuthenticatorStep.signUp ? 1 : 0,
          ),
        );
```

**Comment by ekjotmultani** [comment]
Thanks for the revived interest on this issue, at the very least it seems we should expose the social buttons, if not just allow this social sign in only flow on the Authenticator, we are looking back into this

**Comment by markvp** [comment]
I think that this ties in with https://github.com/aws-amplify/amplify-backend/issues/1656
